bin/cliain/src/main.rs

```diff
@@ -13,38 +17,175 @@ struct Config {
     pub node: String,
 
     /// The seed of the key to use for signing calls
+    /// If not given, the root account is assumed and user is prompted to provide root seed
     #[clap(long)]
-    pub seed: String,
+    pub seed: Option<String>,
 
-    /// Specific command to execute
+    /// Specific command that executes either a signed transaction or is an auxiliary command
     #[clap(subcommand)]
     pub command: Command,
 }
 
 #[derive(Debug, Clone, Subcommand)]
 enum Command {
+    /// Staking call to bond stash with controller
+    Bond {
+        /// Seed for a stash account
+        #[clap(long)]
+        stash_seed: String,
+
+        /// SS58 id of the controller account
+        #[clap(long)]
+        controller_account: String,
+
+        /// a Stake to bond (in tokens)
+        #[clap(long)]
+        initial_stake_tokens: u32,
+    },
+
     /// Change the validator set for the session after the next
     ChangeValidators {
         /// The new validators
         #[clap(long, value_delimiter = ',')]
         validators: Vec<String>,
     },
+
+    /// Force new era in staking world. Requires sudo.
+    ForceNewEra,
+
     /// Associate the node with a specific staking account.
     PrepareKeys,
+
+    /// Call rotate_keys() RPC call and prints them to stdout
+    RotateKeys,
+
+    /// Sets given keys for origin controller
+    SetKeys {
+        /// 64 byte hex encoded string in form 0xaabbcc..
+        /// where aabbcc...  must be exactly 128 characters long
+        #[clap(long)]
+        new_keys: String,
+
+        /// Seed for a controller account which signes set_keys tx
+        #[clap(long)]
+        controller_seed: String,
+    },
+
+    /// Command to convert given seed to SS58 Account id
+    SeedToSS58 {
+        /// Seed of the account to convert to SS58 public key to
+        #[clap(long)]
+        seed: String,
```

Comment:
I'm pretty sure you can just use the global `--seed` flag and then there should be no problem with treating this as a normal subcommand?

Reply:
Right, I was blind, changed.